### PR TITLE
chore: Update app K8s dependencies

### LINKED ISSUE #635: CVE: PRISMA-2022-0227 in emicklei/go-restful/v3
Twistlock reports the following vulnerability with aws-iam-authenticator:

**Package:** github.com/emicklei/go-restful/v3
**CVE:** [PRISMA-2022-0227](https://github.com/emicklei/go-restful/issues/497)
**Fix Status:** Fixed in:v3.10.0
**Impacted versions:** <v3.10.0

> github.com/emicklei/go-restful/v3 module prior to v3.10.0 is vulnerable to Authentication Bypass by Primary Weakness. There is an inconsistency in how go-restful parses URL paths. This inconsistency could lead to several security check bypass in a complex system.

---

Currently aws-iam-authenticator is bringing in `v3.9.0` as an **indirect** dependency:

https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/4aec898c62bd8277994e224184b21822bb40a70a/go.mod#L31

go-restful v3.10.0 introduced a regression, but v3.11.0 [fixes the vulnerability](https://github.com/emicklei/go-restful/issues/521#issuecomment-1684876345) and also [fixes the regression](https://github.com/emicklei/go-restful/issues/519#issuecomment-1684876181).

Suggestion: `go.mod` should be updated when possible.

### PR DESCRIPTION
**Pull request by bryantbiggs**
**What this PR does / why we need it**:
- Updates application's K8s dependencies
- Closes #627
- Resolves #635 
- Relates #611

This is part of splitting out changes from #684 in smaller increments. `make test` passes successfully locally on this change set

```
...
--- PASS: TestGetIdentityFromSTSResponse (0.00s)
    --- PASS: TestGetIdentityFromSTSResponse/Root_User (0.00s)
    --- PASS: TestGetIdentityFromSTSResponse/User (0.00s)
    --- PASS: TestGetIdentityFromSTSResponse/Role (0.00s)
    --- PASS: TestGetIdentityFromSTSResponse/Federated_User (0.00s)
PASS
coverage: 75.7% of statements
ok  	sigs.k8s.io/aws-iam-authenticator/pkg/token	1.442s	coverage: 75.7% of statements
go tool cover -html=coverage.out -o coverage.html
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by bryantbiggs** [comment]
struggling to figure out this gingko error:

```
 Unknown Decorator
var _ = Describe("[apiserver] the apiserver", framework.WithDisruptive(), func() {
/home/prow/go/src/github.com/kubernetes-sigs/aws-iam-authenticator/tests/e2e/apiserver_test.go:44
  [Container] node was passed an unknown decorator:
  'framework.label{parts:[]string{"Disruptive"}, extra:""}'
  Learn more at: http://onsi.github.io/ginkgo/#node-decorators-overview
```

File: tests/e2e/apiserver_test.go [hunk 1]
```diff
@@ -41,33 +38,27 @@ const (
 	testTimeout  = 300 * time.Second
 )
 
-var _ = Describe("[apiserver] [Disruptive] the apiserver", func() {
-	var (
-		cs  clientset.Interface
-		cfg *restclientset.Config
-	)
+var _ = Describe("the apiserver", framework.WithDisruptive(), func() {
```
Comment (dims) [inline tests/e2e/apiserver_test.go:41]: `SIGDescribe` perhaps?

**Comment by bryantbiggs** [comment]
cc @jyotimahapatra if you get a moment to re-review, thank you!

File: tests/e2e/apiserver_test.go [hunk 2]
```diff
@@ -41,31 +38,27 @@ const (
...
-		cfg *restclientset.Config
-	)
+var SIGDescribe = framework.SIGDescribe("api-machinery")
+
+var _ = SIGDescribe("apiserver", framework.WithDisruptive(), func() {
+	f := framework.NewDefaultFramework("apiserver")
 
 	When("the manifest changes", func() {
 		BeforeEach(func() {
-			cfg, _ = clientcmd.BuildConfigFromFlags("", framework.TestContext.KubeConfig)
```
Comment (jyotimahapatra) [inline tests/e2e/apiserver_test.go:52]: for familiarity: is this not needed because `f := framework.NewDefaultFramework("apiserver")` is added above?

Comment (bryantbiggs) [reply tests/e2e/apiserver_test.go:52]: yes! I'm not well versed in the testing framework but upstream has made several ergonomic improvements so now the config and client set can be extracted from the framework itself

File: .gitignore [hunk 3]
```diff
@@ -69,3 +69,6 @@ cscope.*
 # coverage.out file
 coverage.out
 coverage.html
+
+# multi-module workspaces
+go.work*
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply .gitignore:74]: to be honest, I don't know the full implications of adding this to the project. but my IDE was failing to resolve paths without the multi-module workspace setup so I added it to work locally but that creates a `go.work` (that looks like below) and a `go.work.sum`. I'm happy to go any way with this - revert the change to the gitignore, keep it, add the `go.work*` files, etc.

```go
// go.work
go 1.21

use (
	./tests/e2e
	./tests/integration
)
```

**Comment by bryantbiggs** [comment]
@jyotimahapatra let me know if there are any other comments or feedback that I can help address, thank you!

File: .gitignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply .gitignore:74]: based on https://github.com/kubernetes/enhancements/issues/4402 - perhaps we do not want to ignore the go workspaces and instead start utilizing them?

**Pull request merged**